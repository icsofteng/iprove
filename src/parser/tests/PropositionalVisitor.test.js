import {parse} from '../'

test("Visitor Test implies", ()=> {
  expect(parse("P implies Q")).toMatchSnapshot()
})

test("Visitor Test and", ()=> {
  expect(parse("P and Q")).toMatchSnapshot()
})

test("Visitor Test iff", ()=> {
  expect(parse("P iff Q")).toMatchSnapshot()
})

test("Visitor Test or", ()=> {
  expect(parse("P or Q")).toMatchSnapshot()
})

test("Visitor Test not", ()=> {
  expect(parse("not P")).toMatchSnapshot()
})

test("Visitor Test literals", ()=> {
  expect(parse("P")).toMatchSnapshot()
  expect(parse("true")).toMatchSnapshot()
  expect(parse("false")).toMatchSnapshot()
})

test("Visitor Integrated Test", () => {
  expect(parse("(P and Q) implies (not X or Y)")).toMatchSnapshot()
})
