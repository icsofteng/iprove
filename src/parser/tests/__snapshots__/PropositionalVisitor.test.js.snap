// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor Integrated Test 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "paren",
        "value": Object {
          "lhs": Object {
            "type": "identifier",
            "value": "P",
            "varType": "Bool",
          },
          "rhs": Object {
            "type": "identifier",
            "value": "Q",
            "varType": "Bool",
          },
          "symbol": "and",
          "type": "binary",
        },
      },
      "rhs": Object {
        "type": "paren",
        "value": Object {
          "lhs": Object {
            "symbol": "not",
            "type": "unary",
            "value": Object {
              "type": "identifier",
              "value": "X",
              "varType": "Bool",
            },
          },
          "rhs": Object {
            "type": "identifier",
            "value": "Y",
            "varType": "Bool",
          },
          "symbol": "or",
          "type": "binary",
        },
      },
      "symbol": "implies",
      "type": "binary",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "Y",
      "varType": "Bool",
    },
    Object {
      "value": "X",
      "varType": "Bool",
    },
    Object {
      "value": "Q",
      "varType": "Bool",
    },
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test and 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "identifier",
        "value": "P",
        "varType": "Bool",
      },
      "rhs": Object {
        "type": "identifier",
        "value": "Q",
        "varType": "Bool",
      },
      "symbol": "and",
      "type": "binary",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "Q",
      "varType": "Bool",
    },
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test iff 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "identifier",
        "value": "P",
        "varType": "Bool",
      },
      "rhs": Object {
        "type": "identifier",
        "value": "Q",
        "varType": "Bool",
      },
      "symbol": "iff",
      "type": "binary",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "Q",
      "varType": "Bool",
    },
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test implies 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "identifier",
        "value": "P",
        "varType": "Bool",
      },
      "rhs": Object {
        "type": "identifier",
        "value": "Q",
        "varType": "Bool",
      },
      "symbol": "implies",
      "type": "binary",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "Q",
      "varType": "Bool",
    },
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test literals 1`] = `
Object {
  "ast": Array [
    Object {
      "type": "identifier",
      "value": "P",
      "varType": "Bool",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test literals 2`] = `
Object {
  "ast": Array [
    Object {
      "type": "true",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test literals 3`] = `
Object {
  "ast": Array [
    Object {
      "type": "false",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test not 1`] = `
Object {
  "ast": Array [
    Object {
      "symbol": "not",
      "type": "unary",
      "value": Object {
        "type": "identifier",
        "value": "P",
        "varType": "Bool",
      },
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;

exports[`Visitor Test or 1`] = `
Object {
  "ast": Array [
    Object {
      "lhs": Object {
        "type": "identifier",
        "value": "P",
        "varType": "Bool",
      },
      "rhs": Object {
        "type": "identifier",
        "value": "Q",
        "varType": "Bool",
      },
      "symbol": "or",
      "type": "binary",
    },
  ],
  "errors": false,
  "functions": Array [],
  "identifiers": Array [
    Object {
      "value": "Q",
      "varType": "Bool",
    },
    Object {
      "value": "P",
      "varType": "Bool",
    },
  ],
  "relations": Array [],
  "types": Array [],
}
`;
